cli/run/importConfigPlugin.ts

```diff
@@ -0,0 +1,26 @@
+import { resolve } from 'path';
+
+export default async function importCustomConfigPlugin(
+	configPath: string,
+	customPlugin: any,
+	extension: string
+) {
+	let plugin = customPlugin;
+	if (!plugin || plugin === true) {
```

Comment:
I think it is slightly awkward to support `--customPlugin` without any parameters and make it have the effect of adding the TypeScript plugin.

So this is indeed roughly what I originally envisioned (except for the added parameter), but did you give some thought of my suggestion of reusing the existing plugin logic? It would provide consistent syntax with the regular `--plugin` option and even allow ad-hoc plugins form the command line if someone desires.

Here is how it would work:

```
// in loadConfigFile.ts
//...
async function loadConfigFile(
	fileName: string,
	commandOptions: any
): Promise<GenericConfigObject[]> {
	const extension = path.extname(fileName);
	const configFileExport =
		!commandOptions.configPlugun && extension === '.mjs' && supportsNativeESM()
			? (await import(pathToFileURL(fileName).href)).default
			: !commandOptions.configPlugun && extension === '.cjs'
			? getDefaultFromCjs(require(fileName))
			: await getDefaultFromTranspiledConfigFile(fileName, commandOptions);
	return getConfigList(configFileExport, commandOptions);
}

//...

async function getDefaultFromTranspiledConfigFile(
	fileName: string,
	commandOptions: any
): Promise<unknown> {
	const warnings = batchWarnings();
	const inputOptions = {
		external: (id: string) =>
			(id[0] !== '.' && !path.isAbsolute(id)) || id.slice(-5, id.length) === '.json',
		input: fileName,
		onwarn: warnings.add,
		plugins: [],
		treeshake: false
	};
	addCommandPluginsToInputOptions(inputOptions, commandOptions, 'configPlugin');
	const bundle = await rollup.rollup(inputOptions);
	if (!commandOptions.silent && warnings.count > 0) {
//...

// In commandPlugins.ts
export function addCommandPluginsToInputOptions(inputOptions: InputOptions, command: any, pluginOption = 'plugin') {
	if (command.stdin !== false) {
		inputOptions.plugins!.push(stdinPlugin(command.stdin));
	}
	if (command.waitForBundleInput === true) {
		inputOptions.plugins!.push(waitForInputPlugin());
	}
	const commandPlugin = command[pluginOption];
	if (commandPlugin) {
//...
```

This would even support shortening the command to

```js
rollup -c rollup.config.ts --configPlugin typescript
```

as it automatically tries the common plugin prefixes first.

Note that plugins are resolved relative to Rollup, not relative to the config file or current working directory as I first thought, but this would be no different for your implementation. Nice thing is that if we want to improve this later, we would improve it for both the --plugin and the --configPlugin option together.